File: CS/08_Bookmarks/GetPdfChildBookmarks.cs
using Spire.Pdf;
using Spire.Pdf.Bookmarks;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace GetPdfChildBookmarks
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a new Pdf document.
            PdfDocument doc = new PdfDocument();

            //Load the file from disk.
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Pdf_1.pdf");

            //Get bookmarks collections of the PDF file.
            PdfBookmarkCollection bookmarks = doc.Bookmarks;

            string result = "GetPdfChildBookmarks_out.txt";

            //Get Pdf child Bookmarks.
            GetChildBookmarks(bookmarks, result);

            //Launch the file.
            DocumentViewer(result);
        }

        private void GetChildBookmarks(PdfBookmarkCollection bookmarks, string result)
        {
            //Declare a new StringBuilder content
            StringBuilder content = new StringBuilder();

            //Get Pdf child Bookmarks information.
            foreach (PdfBookmark parentBookmark in bookmarks)
            {
                if (parentBookmark.Count > 0)
                {
                    content.AppendLine("Child Bookmarks:");

                    foreach (PdfBookmark childBookmark in parentBookmark)
                    {
                        //Get the title
                        content.AppendLine(childBookmark.Title);

                        //Get the color.
                        string color = childBookmark.Color.ToString();
                        content.AppendLine(color);

                        //Get the text style.
                        string textStyle = childBookmark.DisplayStyle.ToString();
                        content.AppendLine(textStyle);
                    }
                }

                //Save to file.
                File.WriteAllText(result, content.ToString());
            }
        }

        private void DocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/08_Bookmarks/GetPdfChildBookmarks.vb
Imports Spire.Pdf
Imports Spire.Pdf.Bookmarks
Imports System.IO
Imports System.Text

Namespace GetPdfChildBookmarks
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument instance
			Dim doc As New PdfDocument()

			' Load an existing PDF document from the specified file path
			doc.LoadFromFile("..\..\..\..\..\..\Data\Template_Pdf_1.pdf")

			' Get the collection of bookmarks from the document
			Dim bookmarks As PdfBookmarkCollection = doc.Bookmarks

			' Specify the output file name
			Dim result As String = "GetPdfChildBookmarks_out.txt"

			' Retrieve and save the child bookmark information recursively
			GetChildBookmarks(bookmarks, result)

			' Close the document
			doc.Close()

			' Launch the file
			DocumentViewer(result)
		End Sub
		Private Sub GetChildBookmarks(ByVal bookmarks As PdfBookmarkCollection, ByVal result As String)
			' Create a StringBuilder to hold the bookmark content
			Dim content As New StringBuilder()

			' Iterate through each parent bookmark in the collection
			For Each parentBookmark As PdfBookmark In bookmarks

				' Check if the parent bookmark has any child bookmarks
				If parentBookmark.Count > 0 Then
					content.AppendLine("Child Bookmarks:")

					' Iterate through each child bookmark within the parent bookmark
					For Each childBookmark As PdfBookmark In parentBookmark

						' Append the title of the child bookmark
						content.AppendLine(childBookmark.Title)

						' Get the color of the child bookmark and append it
						Dim color As String = childBookmark.Color.ToString()
						content.AppendLine(color)

						' Get the display style of the child bookmark and append it
						Dim textStyle As String = childBookmark.DisplayStyle.ToString()
						content.AppendLine(textStyle)
					Next childBookmark
				End If

				' Write the bookmark content to the specified file
				File.WriteAllText(result, content.ToString())
			Next parentBookmark
		End Sub
		Private Sub DocumentViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace
